src/status_im/contexts/communities/events.cljs

```diff
@@ -163,12 +163,15 @@
 (defn initialize-permission-addresses
   [{:keys [db]} [community-id]]
   (when community-id
-    (let [accounts        (get-in db [:wallet :accounts])
-          sorted-accounts (sort-by :position (vals accounts))
-          addresses       (set (map :address sorted-accounts))]
+    (let [accounts         (get-in db [:wallet :accounts])
+          sorted-accounts  (sort-by :position (vals accounts))
+          watched-accounts (remove #(:watch-only? %) sorted-accounts)
+          addresses        (set (map :address watched-accounts))]
       {:db (update-in db
                       [:communities community-id]
                       assoc
+                      :previous-share-all-addresses? true
+                      :share-all-addresses?          true
```

Comment:
I understand that the community object is not the right place to store the UI state for sharing all current and future addresses. If you have no objections, I will keep it here for now and move it out of the community object in a separate PR, along with the other UI states for the addresses on the permissions screen.

Related issue: https://github.com/status-im/status-mobile/issues/18844